File: baserecycleradapter/src/main/java/com/gigigo/baserecycleradapter/debouncedlisteners/DebouncedOnLongClickListener.java
package com.gigigo.baserecycleradapter.debouncedlisteners;

import android.view.View;

import com.gigigo.baserecycleradapter.viewholder.BaseViewHolder;

public abstract class DebouncedOnLongClickListener
        implements DebouncedListener, BaseViewHolder.OnItemLongClickListener {

    private final DebouncedClickHandler debouncedClickHandler;

    protected DebouncedOnLongClickListener() {
        this.debouncedClickHandler = new DebouncedClickHandler(this);
    }

    @Override
    public boolean onItemLongClicked(int position, View view) {
        return debouncedClickHandler.invokeDebouncedClick(position, view);
    }
}

File: baserecycleradapter/src/main/java/com/gigigo/baserecycleradapter/debouncedlisteners/DebouncedOnLongClickListener.kt
package com.gigigo.baserecycleradapter.debouncedlisteners

import android.view.View

import com.gigigo.baserecycleradapter.viewholder.BaseViewHolder

internal abstract class DebouncedOnLongClickListener protected constructor() : DebouncedListener,
    BaseViewHolder.OnItemLongClickListener {

    @Suppress("LeakingThis")
    private val debouncedClickHandler: DebouncedClickHandler = DebouncedClickHandler(this)

    override fun onItemLongClicked(position: Int, view: View): Boolean =
        debouncedClickHandler.invokeDebouncedClick(position, view)
}
